zerver/lib/actions.py

```diff
@@ -4985,23 +4986,42 @@ def get_status_dict(requesting_user_profile: UserProfile) -> Dict[str, Dict[str,
     return UserPresence.get_status_dict_by_realm(requesting_user_profile.realm_id)
 
 def get_cross_realm_dicts() -> List[Dict[str, Any]]:
+    # Using the deferred import to avoid circular import error.
+    from zerver.lib.events import format_user_row
+
     users = bulk_get_users(list(settings.CROSS_REALM_BOT_EMAILS), None,
                            base_query=UserProfile.objects.filter(
-                               realm__string_id=settings.SYSTEM_BOT_REALM)).values()
-    return [{'email': user.email,
-             'user_id': user.id,
-             'is_admin': user.is_realm_admin,
-             'is_bot': user.is_bot,
-             'avatar_url': avatar_url(user),
-             'timezone': user.timezone,
-             'date_joined': user.date_joined.isoformat(),
-             'bot_owner_id': None,
-             'full_name': user.full_name}
-            for user in users
-            # Important: We filter here, is addition to in
-            # `base_query`, because of how bulk_get_users shares its
-            # cache with other UserProfile caches.
-            if user.realm.string_id == settings.SYSTEM_BOT_REALM]
+                           realm__string_id=settings.SYSTEM_BOT_REALM)).values()
+    result = []
+    for user in users:
+        # Important: We filter here, is addition to in
+        # `base_query`, because of how bulk_get_users shares its
+        # cache with other UserProfile caches.
+        if user.realm.string_id == settings.SYSTEM_BOT_REALM:
+            result.append(format_user_row(user.realm,
+                                          user_profile=user,
+                                          row=dict(email=user.email,
+                                                   id=user.id,
+                                                   user_profile=user,
```

Comment:
At first I thought at that this was a security issue because user_profile is supposed to be that of the requesting user and not the user of the loop - this would have lead to leaking data (emails of the admin) at the 
```python3
        if (realm.email_address_visibility == Realm.EMAIL_ADDRESS_VISIBILITY_ADMINS and
                user_profile.is_realm_admin):
            result['delivery_email'] = row['delivery_email']
```
part. But then I realized that this only looks at the cross realm bots, so there's no real security issue here, but I wanted to mention all of this anyways just because it's something that could be easily missed and should be heeded in general. Maybe adding a comment around here acknowledging this would be a good idea for others later.

Reply:
I think it's best to be really clear about how we're naming things; calling the parameter passed in `acting_user` rather than `user_profile` is technique we use elsewhere where we want to have a loop over UserProfile objects have that `user_profile` variable name.

Reply:
So , should I rename that parameter to acting_user in format_user_row ?

Reply:
Yup. Change the parameter name itself.